src/Microsoft.Maintainability.Analyzers/Core/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -37,6 +43,205 @@ public override void Initialize(AnalysisContext analysisContext)
 
```

Comment:
We should try to enable concurrent execution for this analyzer - basically uncomment the line `//analysisContext.EnableConcurrentExecution();` and then use concurrent sets instead of plain lists to hold instantiatedTypes and internalTypes.